### PR TITLE
Narrow e2e pre-check on scheduler predicates

### PR DESCRIPTION
**Pull request by liggitt**
WaitForAllNodesHealthy actually checks for optional add-ons running on all nodes. The scheduler tests only care about node readiness, so just check for that.

Hit this running scheduler e2e tests downstream on clusters that do not deploy those optional add-ons. The test would just hang and fail unnecessarily

/sig scheduling

```release-note
NONE
```

### CODE REVIEW

**Comment by liggitt** [comment]
cc @kubernetes/sig-scheduling-pr-reviews

File: test/e2e/scheduling/predicates.go [hunk 1]
```diff
@@ -90,20 +89,6 @@ var _ = SIGDescribe("SchedulerPredicates [Serial]", func() {
 		// Every test case in this suite assumes that cluster add-on pods stay stable and
 		// cannot be run in parallel with any other test that touches Nodes or Pods.
 		// It is so because we need to have precise control on what's running in the cluster.
-		systemPods, err := framework.GetPodsInNamespace(cs, ns, ignoreLabels)
```
Comment (wgliang) [inline test/e2e/scheduling/predicates.go]: ignoreLabels declared but will not used.

Comment (soltysh) [reply test/e2e/scheduling/predicates.go]: The above comment doesn't make sense anymore when you remove this code, as well.

Comment (timothysc) [reply test/e2e/scheduling/predicates.go]: Also, do you have a list of what the pre-condition was waiting for that is missing on openshift?

**Comment by dims** [comment]
looks like a bunch of failed jobs ... are they legit?

**Comment by deads2k** [comment]
@liggitt nag

**Comment by liggitt** [comment]
hoisting comment from https://github.com/kubernetes/kubernetes/pull/61718#discussion_r194599699:

> do you have a list of what the pre-condition was waiting for that is missing on openshift

these were the hardcoded pods this test required to consider a node healthy that do not match all deployments:

https://github.com/kubernetes/kubernetes/blob/6decdaa775f5feb480d06e05c049fa564187f30e/test/e2e/framework/util.go#L221-L226

also, strangely, this was looking for pods in the current test namespace that were running on non-master nodes, then requiring at least that many pods (just comparing counts) to be running in the kube-system namespace.

**Review by bsalamat — APPROVED** [review]
/lgtm

Thanks, Jordan!

**Pull request merged**